crates/outbound-http/src/host_impl.rs

```diff
@@ -63,9 +63,14 @@ impl outbound_http::Host for OutboundHttp {
 
             let req_url = reqwest::Url::parse(&abs_url).map_err(|_| HttpError::InvalidUrl)?;
 
-            let headers = request_headers(req.headers).map_err(|_| HttpError::RuntimeError)?;
+            let mut headers = request_headers(req.headers).map_err(|_| HttpError::RuntimeError)?;
             let body = req.body.unwrap_or_default().to_vec();
 
+            // if no content length, set it
+            if headers.get("content-length").is_none() && !body.is_empty() && headers.get("transfer-encoding").is_none() {
+                headers.append("content-length", body.len().into());
+            }
```

Comment:
Constants prevent typos:

```rust
use http::header::{CONTENT_LENGTH, TRANSFER_ENCODING};
```

`contains_key` is a little more idiomatic/comprehensible:
```suggestion
            // set content length if necessary
            if !body.is_empty()
                 && !headers.contains_key(CONTENT_LENGTH)
                 && !headers.contains_key(TRANSFER_ENCODING)
             {
                 headers.insert(CONTENT_LENGTH, body.len().into());
             }
  
```